core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -300,6 +304,7 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
       // Ensures writes to the data container use the right consistent hash
       // No need for a try/finally block, since it's just an assignment
       stateTransferLock.acquireExclusiveTopologyLock();
+      triangleOrderManager.updateCacheTopology(cacheTopology);
```

Comment:
What happens with commands with the old topology id that were queued at this point, if they manage to run before `this.cacheTopology = cacheTopology`? Granted, that window is likely to be very very short, but in theory we could be executing some commands in the wrong order.